qadence2_expressions/core/constructors.py

```diff
@@ -1,41 +1,92 @@
 from __future__ import annotations
 
+from functools import wraps
 from typing import Any, Callable
 
 from .environment import Environment
 from .expression import Expression
 from .support import Support
 
 
+def with_repr(repr_func: Callable) -> Callable:
+    """Decorator to give a dynamic __repr__ to a function based on its arguments."""
+
+    class CallableWithRepr:
+        def __init__(
+            self,
+            func: Callable,
+            *args: Any,
+            **kwargs: Any,
+        ) -> None:
+            self.func: Callable = func
+            self.args = args
+            self.kwargs = kwargs
+
+        def __call__(self, *args: Any, **kwargs: Any) -> Any:
+            # Call the original function with provided arguments
+            return self.func(*args, **kwargs)
+
+        def __repr__(self) -> str:
+            # Generate a custom repr using the provided repr_func
+            return str(repr_func(self.func, *self.args, **self.kwargs))
+
+    @wraps(repr_func)
+    def decorator(func: Callable) -> Callable:
+        # Return a wrapped CallableWithRepr instance
+        return lambda *args, **kwargs: CallableWithRepr(func(*args, **kwargs), *args, **kwargs)
+
+    return decorator
+
+
 def value(x: complex | float | int) -> Expression:
     """Create a numerical expression from the value `x`.
 
     Args:
-        x: Any numerical value.
+        x (complex | float | int): Any numerical value.
+
+    Returns:
+        Expression: An expression of type value.
+
+    Raises:
+        TypeError: If the argument is non-numerical.
     """
+    if not isinstance(x, (complex, float, int)):
+        raise TypeError(
+            "Input to 'value' constructor must be of type 'complex', 'float' or 'int'. "
+            f"Got {type(x)}."
+        )
 
     return Expression.value(x)
 
 
 def promote(x: Expression | complex | float | int) -> Expression:
-    """Used to type cast inputs as expressions.
+    """Type cast inputs as value type expressions.
 
     Args:
-        x: A valid expression or numerical value. Numerical values are converted into `Value(x)``
-            expressions.
+        x (Expression | complex | float | int): A valid expression or numerical value.
+         Numerical values are converted into `Value(x)` expressions.
+
+    Returns:
+        Expression: A value type or expression.
     """
 
     return value(x) if not isinstance(x, Expression) else x
 
 
 def symbol(identifier: str, **attributes: Any) -> Expression:
-    """Create a new symbol from the `identfier` if it is not protected.
+    """Create a new symbol from the `identifier` if not protected.
 
     Args:
-        identifier: Symbol's name.
+        identifier (str): Symbol's name.
```

Comment:
Someone told me mkdocs doesn't like the parenthesis type addition. Maybe check if the formatting in mkdocs looks good.

Reply:
Isn't the Google standard?

Reply:
It seems it is and I checked the docs but it I haven't developed it yet. This is for the next PR.